--- faultyVersion.py
+++ fixed.py
@@ -10,7 +10,6 @@
         N//=x
 if N>1:
     print(0)
-    
 else:
     a,b,c=cnt
     dp=[[[0]*(c+1) for j in range(b+1)] for i in range(a+1)]
@@ -28,7 +27,7 @@
                 if i+2<=a:
                     dp[i+2][j][k]+=dp[i][j][k]*inv%mod
                 if k+1<=c:
-                    dp[i][j][k+1]=dp[i][j][k]*inv%mod
+                    dp[i][j][k+1]+=dp[i][j][k]*inv%mod
                 if i+1<=a and j+1<=b:
                     dp[i+1][j+1][k]+=dp[i][j][k]*inv%mod
     print(dp[a][b][c])